firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/annotations/KDocumentId.kt

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx.annotations
+
+import kotlinx.serialization.SerialInfo
+
+/**
+ * Annotation used to mark a @Serializable object's property to be automatically populated with the
+ * document's ID when the object is created from a Cloud Firestore document.
+ * 
+ * <p>When using a @Serializable object to write to a document, the property annotated by
```

Comment:
Fields annotated with this class can be understood as `@Transient` in Serializer "jargon", right? It would be useful to make the analogy here, and in server timestamp.

Reply:
Actually @Transient is the equivalent to Firestore's Java @Exclude; and @Transient is provided by Kotlin serialization core lib by default, I can directly use @Transient in my code base.

I have updated my comments and made the analogy between:
1. @KDocumentId and @DocumentId
2. @KServerTimestamp and @ServerTimestamp
3. @KThrow/IgnoreExtraProperties and Json { ignoreUnknownKeys =false/true }